aws/src/main/java/org/apache/iceberg/aws/s3/S3InputStream.java

```diff
@@ -81,6 +89,8 @@ public int read() throws IOException {
 
     pos += 1;
     next += 1;
+    readBytes.increment(1L);
```

Comment:
Do we want to update the metrics every time here? This feels like it may be expensive although I obviously would want to check benchmarks to be sure. I think a lot of systems tend to just update metrics every X operations and then do a final update in the "close" method

Reply:
Yeah, this method shouldn't really be called at all (if you go back to the original implementation we were considering not even including it).  This is more for completeness of the InputStream api as opposed to trying to be performant.

tl;dr don't use this method :)